src/mongoose_hooks.erl

```diff
@@ -455,16 +456,24 @@ xmpp_stanza_dropped(Acc, From, To, Packet) ->
 
 %% C2S related hooks
 
--spec c2s_broadcast_recipients(State, Type, From, Packet) -> Result when
+-spec get_pep_recipients(State, Feature) -> Result when
     State :: mongoose_c2s:data(),
-    Type :: {atom(), any()},
-    From :: jid:jid(),
-    Packet :: exml:element(),
+    Feature :: binary(),
     Result :: [jid:simple_jid()].
-c2s_broadcast_recipients(State, Type, From, Packet) ->
-    Params = #{state => State, type => Type, from => From, packet => Packet},
+get_pep_recipients(State, Feature) ->
+    Params = #{state => State, feature => Feature},
```

Comment:
Here I would avoid calling it just state, because the terminology for state machines is overloaded: "state" refers to the state the machine is, in its fsm diagram, not to the state of its memory, i.e., the machine's data. Initially terms were a bit mixed because we're way too used to call state the data, that's why I have some variables called `StateData`, but then for consistency I tried to moved to `C2SState` and `C2SData`.

Same comment applies to the hook below.